include/gz/sim/SystemPriorityConstants.hh

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef GZ_SIM_SYSTEMPRIORITYCONSTANTS_HH_
+#define GZ_SIM_SYSTEMPRIORITYCONSTANTS_HH_
+
+#include <gz/sim/System.hh>
+
+namespace gz::sim::systems
+{
+  /// \brief Default priority value for the UserCommands system, with a very
+  /// negative value to indicate that it should run before most systems in
+  /// order to modify the ECM in response to user commands received over
+  /// gz-transport. This is especially important for user commands that add
+  /// objects to the world.
+  constexpr System::PriorityType kUserCommandsPriority = -10000;
+
+  /// \brief A suggested priority value for a system that should execute before
+  /// the Physics system.
+  constexpr System::PriorityType kPrePhysicsPriority = -128;
+
+  /// \brief Default priority value for the Physics system, with a negative
+  /// value ensuring that it will run before systems with priority
+  /// System::kDefaultPriority.
+  constexpr System::PriorityType kPhysicsPriority = -64;
+
+  /// \brief A suggested priority value for sensor systems that should execute
+  /// after the Physics system but before the systems with priority
+  /// System::kDefaultPriority.
+  constexpr System::PriorityType kPostPhysicsSensorPriority = -32;
+}
```

Comment:
move `kDefaultPriority` from System.hh to this file so all constants can be found in the same place?

Reply:
I'm adding a file called `Constants.hh` in https://github.com/gazebosim/gz-sim/pull/2497/files#diff-01974b0165a380ef2dd215a32374bd9192d833c34ca887784b82d89a4605735e. We could also put it in there. I don't have a strong opinion.

Reply:
should I define all these constants in `System.hh`?

Reply:
I moved all the constants to `System.hh` in https://github.com/gazebosim/gz-sim/pull/2500/commits/5a4f1701eef8b544bd20e3bb303ceb732737f406